test/hotspot/jtreg/runtime/ErrorHandling/PrintVMInfoAtExitTest.java

```diff
@@ -52,18 +48,20 @@ public static void main(String[] args) throws Exception {
             "-XX:CompressedClassSpaceSize=256m",
             "-version");
 
+    // How many kb of committed memory we expect in the NMT summary.
+    int committed_kb = 65536;
     OutputAnalyzer output_detail = new OutputAnalyzer(pb.start());
     output_detail.shouldContain("# JRE version:");
     output_detail.shouldContain("--  S U M M A R Y --");
     output_detail.shouldContain("Command Line: -Xmx64M -Xms64M -XX:-CreateCoredumpOnCrash -XX:+UnlockDiagnosticVMOptions -XX:+PrintVMInfoAtExit -XX:NativeMemoryTracking=summary -XX:CompressedClassSpaceSize=256m");
     output_detail.shouldContain("Native Memory Tracking:");
-    WhiteBox wb = WhiteBox.getWhiteBox();
-    if (wb.isAsanEnabled()) {
-        // the reserved value can be influenced by asan
-        output_detail.shouldContain("Java Heap (reserved=");
-        output_detail.shouldContain(", committed=65536KB)");
-    } else {
-        output_detail.shouldContain("Java Heap (reserved=65536KB, committed=65536KB)");
+    // Make sure the heap summary is present.
+    output_detail.shouldMatch("Java Heap \\(reserved=[0-9]+KB, committed=" + committed_kb + "KB\\)");
+    // Check reserved >= committed.
+    String reserved_kb_string = output_detail.firstMatch("reserved=([0-9]+)KB, committed=" + committed_kb + "KB", 1);
```

Comment:
This will likely always match the "Java Heap" line, but should we be more explicit and add "Java Heap" like this?
```suggestion
    String reserved_kb_string = output_detail.firstMatch("Java Heap \\(reserved=([0-9]+)KB, committed=" + committed_kb + "KB\\)", 1);
```

Reply:
Good shout.